### ISSUE TITLE
Add statusCode into err message

### ISSUE THREAD

**Issue by janholusa** [composer/composer#10830]
Since error messages are part of some if statements - it might be useful to provide also row status code to user which might be a good guide for a debugging.

Example what happened to me:
Using `composer install` in jenkins I obtained something like this:
```
In AuthHelper.php line 170:
                                                                               
  The '<https path to private repository>' URL required authentication.      
  You must be using the interactive console to authenticate
```
First - shock for me that it seems to not possible to use `composer install` with private repository in jenkins.
But it is nonsense, so I checked AuthHelper.php (here is latest version, I checked version I use - that is why line differs):
https://github.com/composer/composer/blob/203ec4b485e43f77a9aa8e5064e72d1c67f96543/src/Composer/Util/AuthHelper.php#L195
Found that it is actually **401** which means password might be incorrect. 
I updated password and it was fixed.

What do you think?

**Comment by Seldaek** [comment]
Yes probably would be valuable to add the status code for clarity 👍🏻

**Comment by Seldaek** [comment]
Well I messed up and put two commits in one, but anyway it's fixed by 5c68c2c86d837682e29562ea5a87871b67a57b85
